[WIP] Reconfig fixes, requires redeploy slave software:

mariangemarcano opened the pull request:
[WIP] Reconfig fixes, requires redeploy slave software: 
- Requeue running builds when slaves gets detached from builder
- Update slave software to skip _ackFailed if the builder gets detached
- Refactor stopBuild to build_status
- Return the interrupt deferred to run actions async
- Builder handles reconfiguration of slaves and detached the ones that are no longer used
- Make sure we retry builds after build request are resubmitted

mariangemarcano committed:
[WIP] Reconfig fixes, requires redeploy slave software:
- Requeue running builds when slaves gets detached from builder
- Update slave software to skip _ackFailed if the builder gets detached
- Refactor stopBuild to build_status
- Return the interrupt deferred to run actions async
- Builder handles reconfiguration of slaves and detached the ones that are no longer used
- Make sure we retry builds after build request are resubmitted

BarryUnity commented on master/buildbot/status/build.py:
```diff
@@ -520,6 +520,18 @@ def checkLogfiles(self):
+        if bldrc:
+            bldc = bldrc.getBuild(self.getNumber())
+            if bldc:
+                yield bldc.stopBuild(reason=reason, result=result)
```
It's mostly superficial, but I really question your choice in variable names here. Is that because that's what it was before you moved the code?

mariangemarcano replied:
@BarryUnity I didn't change variable names, it was before like this I just move the code to common place

BarryUnity commented on master/buildbot/status/build.py:
```diff
@@ -520,6 +520,18 @@ def checkLogfiles(self):
+            if bldc:
+                yield bldc.stopBuild(reason=reason, result=result)
+
+        defer.succeed(None)
```
Any reason this is always returning Success, regardless of the actual result?

BarryUnity commented on slave/buildslave/bot.py:
```diff
@@ -225,12 +226,16 @@ def commandComplete(self, failure):
         self.command = None
         if not self.running:
             log.msg(" but we weren't running, quitting silently")
-            return
```
This log message doesn't seem accurate anymore.

henriksunity reviewed (approved):
Looks good!

campos-ddc commented:
Approved

mariangemarcano committed:
Removing a slave (during reconfiguration) should disconnect it from the current master, this allows cleanly reattach it afterwards

mariangemarcano committed:
Handle the case where a builder is removed during reconfiguration while running a build

mariangemarcano committed:
Fix: Allow cancel buildrequest in queue even when the builder is removed.
This handle the case when builder is removed during reconfiguration. In this case build status update is skip.

mariangemarcano committed:
Merge branch 'staging' into fixes/reconfiguration

mariangemarcano committed:
Merge branch 'staging' into fixes/reconfiguration

mariangemarcano committed:
Display more user friendly message when builds gets interrupted due to reconfiguration

mariangemarcano committed:
Adjust connection lost message

mariangemarcano commented:
We are putting on hold this changes no reconfiguration supported on this version of katana.
